pkgs/development/compilers/mozart/default.nix

```diff
@@ -1,91 +1,31 @@
-{ lib
-, fetchFromGitHub
-, fetchurl
-, cmake
+{ stdenv
+, lib
+, callPackage
 , unzip
-, makeWrapper
-, boost169
-, pinnedBoost ? boost169
-, llvmPackages
-, llvmPackages_5
-, gmp
 , emacs
-, jre_headless
-, tcl
-, tk
+, makeWrapper
+, installDesktop ? false
 }:
+let
+  mozart2-unwrapped = callPackage ./unwrapped.nix { };
+in
+stdenv.mkDerivation {
 
-let stdenv = llvmPackages.stdenv;
-
-in stdenv.mkDerivation rec {
-  pname = "mozart2";
-  version = "2.0.1";
-  name = "${pname}-${version}";
-
-  src = fetchurl {
-    url = "https://github.com/mozart/mozart2/releases/download/v${version}/${name}-Source.zip";
-    sha256 = "1mad9z5yzzix87cdb05lmif3960vngh180s2mb66cj5gwh5h9dll";
-  };
+  nativeBuildInputs = [ makeWrapper unzip ];
 
-  # This is a workaround to avoid using sbt.
-  # I guess it is acceptable to fetch the bootstrapping compiler in binary form.
-  bootcompiler = fetchurl {
-    url = "https://github.com/layus/mozart2/releases/download/v2.0.0-beta.1/bootcompiler.jar";
-    sha256 = "1hgh1a8hgzgr6781as4c4rc52m2wbazdlw3646s57c719g5xphjz";
-  };
+  propagatedBuildInputs = [ mozart2-unwrapped emacs ];
 
-  patches = [ ./patch-limits.diff ];
+  phases = [ "unpackPhase" "installPhase" "fixupPhase" ];
 
-  postConfigure = ''
-    cp ${bootcompiler} bootcompiler/bootcompiler.jar
+  installPhase = ''
+    install -D opi/emacs/oz $out/bin/oz
+  '' + lib.optionalString installDesktop ''
+    cp -r distrib/share/ $out/share
```

Comment:
The only thing this wrapper should do is to wrap the executable with an emacs, propagating all other files without changes. I recommend using something like

```nix
symlinkJoin {
  inherit (mozart2-unwrapped) name pname version meta;
  paths = [ mozart2-unwrapped ];
  nativeBuildInputs = [ makeWrapper ];
  postBuild = ''
    wrapProgram ...
  '';
}
```

Reply:
I didn't know this function, thank you for pointing it out. But considering that the desktop files are not presented by mozart2-unwrapped, can it still be used?